server/application/application.go

```diff
@@ -206,6 +208,20 @@ func (s *Server) Create(ctx context.Context, q *application.ApplicationCreateReq
 	return updated, nil
 }
 
+// GetNodes returns nodes associated with an application
+func (s* Server) GetNodes(ctx context.Context, q *application.ApplicationQuery) (error) {
+	labelsMap, err := labels.ConvertSelectorToLabelsMap(q.Selector)
+	if err != nil {
+		return err
+	}
+	nodes, err := s.nodeLister.List(labelsMap.AsSelector())
```

Comment:
1. We need to perform filtering of nodes only relevant to the application. This will need to first get the Pods of the application, collect all the node names of the pods, then filter the node list to only those which are relevant to the application.

2. node objects should be redacted to only the information needed to be presented in the UI. It's a security concern to return the full node objects because it reveals things like what images are being run on the node.